### ISSUE TITLE
NA in word changes id to NA after unnest_tokens()

### ISSUE THREAD

**Issue by twedl** [juliasilge/tidytext#82]
Using tidytext to help process names and addresses and such (very helpful, ty👍 🙌 ). Sometimes an ```id``` has as ```NA``` in the field I want to process. ```unnest_tokens()``` correctly says the token is ```NA``` but also replaces the ```id``` with ```NA``` for some reason.
```
df <- tibble(id = c("A", "B"), address = c(NA, "742 Evergreen Terrace")) %>% 
          unnest_tokens(word, address)
df
# A tibble: 4 x 2
     id      word
  <chr>     <chr>
1  <NA>      <NA>
2     B       742
3     B evergreen
4     B   terrace
```

Where one might expect this output: 
```
df
# A tibble: 4 x 2
     id      word
  <chr>     <chr>
1     A      <NA>
2     B       742
3     B evergreen
4     B   terrace
```

This might be uncommon use, and in the meantime I just filter the ```NA``` observations before calling ```unnest_tokens()```, but I thought you might want to be aware of this. Thanks!

Some session info:
```
sessionInfo()
R version 3.4.0 (2017-04-21)
[etc etc etc]
tidyverse_1.1.1 tidytext_0.1.3
```

**Comment by twedl** [comment]
The problem is in line 183 in ```unnest_tokens()```, in the ```unnest_tokens.data.frame()``` function: 

```ret <- ret[ret[[output]] != "", , drop = FALSE]```

The ```!=``` bit sets everything to ```NA``` in the line with an ```NA``` output, but doesn't drop it. Changing this line to

```ret <- ret[!is.na(ret[[output]]), , drop = FALSE]```

drops those lines (any word with ```""``` or ```NA```), if that's the desired output. Removing the line altogether keeps all the ```NA``` lines, drops empty strings (```""```) and also doesn't set the id to ```NA```, if that's the desired output. I'm not sure how those changes affect other use cases, but either one seems to pass the tests (NB: it's my first time using ```testthat```).

I noticed that line is related to these issues but wasn't sure how: [afc24f4cee22ae7cbbf7b697f6cbfada81d5e4e3](https://github.com/juliasilge/tidytext/commit/afc24f4cee22ae7cbbf7b697f6cbfada81d5e4e3) which fixed a prior bug in [c5215b1c4b2766ebd600482fb73f87fedb6f45a2](https://github.com/juliasilge/tidytext/commit/c5215b1c4b2766ebd600482fb73f87fedb6f45a2).

**Comment by juliasilge** [comment]
Thanks so much! 🙏 That was in fact weird behavior.
